### DISCUSSION TITLE
Du Guesclin, une trilogie, published using asciidoctor-web-pdf.

### DISCUSSION THREAD

**Discussion by benoitrolland** [Mogztter/asciidoctor-web-pdf#685] [Show and tell]
3 books of 224, 262 and 251 pages, privatly published using asciidoctor-web-pdf and telling the Story of Du Guesclin, where published on amazon (in french) to get private author copies [see here](https://www.amazon.com/s?k=du+guesclin+une+trilogie)

**Comment by ggrossetie** [comment]
Congratulations, this is awesome!!
Any screenshots (or photos) of the book? I'm curious about your workflow and what were the main pain points? (i.e., what we could do to improve your experience).

Anyway, thanks for sharing 👍🏻

**Comment by benoitrolland** [comment]
### Screenshots: 
![presentationLivre](https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/c679c571-1af4-4517-bb07-7c1181f684ba)

<img width="474" alt="imageCitationEtReference" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/7e9e7297-de7e-4aa3-8d73-7850ff9d35ba">

<img width="433" alt="imageMotsGlossaire" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/3ebfd984-7923-450b-afab-81beae071551">

<img width="518" alt="citationVieuxFrançaisReference" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/ede6c63e-70c2-4a7b-96e1-6c9096a83041">

<img width="534" alt="citationEnluminureVieuxFrançais" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/4cf4c8eb-fe37-482b-9ed2-1963834d576e">

<img width="427" alt="citationVieuxFrançaisTermeDuGlossaireEtDialogueDansReference" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/ed66e9b3-4394-43f2-9653-fbeb7d0a1ed6">

<img width="449" alt="biblio" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/6b1a2de2-7579-4ae3-9bb8-68fcc4b4351e">

<img width="542" alt="Glossaire" src="https://github.com/ggrossetie/asciidoctor-web-pdf/assets/5108268/b518590f-f0c9-4d72-bc96-278b053aea5f">

more details on [amazon preview](https://www.amazon.com/dp/B0C46BD9CW) (clic on image, wait a few seconds, and scroll down).

### Workflow: 
under cygwin bash:
  source="./duguesclin.adoc"
  target="./Du-Guesclin-tome-$VOLNUM.$legalzone.$lang.adoc.pp"
asciidoctor-reducer -a selectTome=$VOLNUM -a lang=$lang -a legalzone=$legalzone -a webpdf -o $target $source
dos2unix $target

source="./Du-Guesclin-tome-$VOLNUM.$legalzone.$lang.adoc.pp"
target="duguesclin-$VOLNUM.$legalzone.$lang.web.pdf"
asciidoctor-web-pdf $source -o $target -a selectTome=$VOLNUM

used qpdf at the end to move the toc at end of the book

### Painpoints: 
 
- Impossible to generate the three volumes into a single pdf because of a timeout issue: htps://github.com/ggrossetie/asciidoctor-web-pdf/issues/661 ( issue said to be solved. Also the problem was solved for me using the same version of asciidoc-web-pdf  using a more powerfull PC ) 
- Impossible to programatically detect when and where some elements (mainly images) can't fit in a page. Requires to check all the book after any little modification !  
- TOC when on several pages not always split well (as a side effect of font size styling ?)
- Some css styling are not took in account to break the pages. (the glossary small font size of volume II should allow it to fit on two pages but it remains on three pages.)
- It is not a simple task to generate the chapter name on top of each right pages
- It is not a simple task to generate page numbers of references listed inside the bibliography and glossary. 
- The [po4a](https://po4a.org/index.php.en) free translation system is not yet fully supporting [asciidoc](https://po4a.org/man/man3/Locale::Po4a::AsciiDoc.3pm.php)
